### ISSUE TITLE
expected depth type ?

### ISSUE THREAD

**Issue by iTomxy** [antocad/FocusOnDepth#19]
Hi,
I adapted your code to train on my own dataset in depth mode as an alternative to DPT, but the loss isn't decreasing, and I found that the output depth map is all zero, which is weird. I found
```python
output_depth = 1-output_depth
```
in [FOD/Predictor.py](https://github.com/antocad/FocusOnDepth/blob/master/FOD/Predictor.py#L57), while the statistics of the depth maps in my dataset is like (*max, min, mean*):
```
77.9375 0.0 1.6585418
82.8125 0.0 10.568505
97.9375 0.0 7.066457
1620.0 0.0 28.10622
203.375 0.0 6.32571
15.890625 0.0 1.8040103
80.625 0.0 11.406916
516.0 0.0 12.47693
5052.0 -3672.0 18.6016
48.75 0.0 8.396343
9136.0 0.0 21.534061
347.5 0.0 8.561709
113.5625 0.0 13.559731
40.875 0.0 9.513472
37.15625 0.0 4.9919477
99.0625 0.0 5.4938717
30.890625 0.0 0.77849835
67.625 0.0 8.707108
42.0 0.0 8.621763
106.1875 0.0 11.275524
74.1875 0.0 7.9050064
92.0 0.0 15.986496
```
So I suspect that the expected depth type is different from the one of my dataset ? What should be the correct depth type ? Like the value range.

thanks

**Comment by wzdnb** [comment]
I have encountered similar problems. Have you solved them？

**Comment by iTomxy** [comment]
> 

I didn't continue on that work. The model failed in learning cuz the ground-truth of my dataset is sparse. But what I did that time is divide the ground-truth depth by a large value to normalise them.

**Comment by wzdnb** [comment]
I found that when I delete the segmentation part and only do depth part, and use the original datasets provided by author, the loss during training will not change basically,and the output depth map is all zero

**Comment by antocad** [comment]
I'll try to answer the questions (sorry It was 1 year ago I don't have the whole code in mind anymore)

- [iTomxy](https://github.com/iTomxy)'s question (sorry for late reply, it can help others):  the range should always be [0;1] for the depth maps. So one can just normalize its input. In the code we used `output_depth = 1-output_depth` because we wanted to reverse the values (0="far" becomes 1, and vice versa; we have done this because of our dataset). 

- [wzdnb](https://github.com/wzdnb)'s issue: Maybe the answer above can help you. If not, the convergence error might be explained by 3 different reasons:

1. The backbone used will load pre-trained weights, while the head will start the training from scratch. having a high learning rate could destroy the weights learned in the backbone and inversely, a low learning rate will prevent the head from being trained.
2. your dataset, (is  your task meaningful ? isn't it too difficult ? like capturing the depth of transparent objects ? )
3. Our code. And in this case, good luck because it's a mess !
